packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -1091,6 +1091,30 @@ void getbuffer256(MachineState& m) {
     ++m.pc;
 }
 
+uint64_t setbuffer_variable_gas_cost(MachineState const& m, uint64_t inc) {
+    // The fixed cost of the the buffer opcodes is applied elsewhere
+    if (m.stack.stacksize() < 3) {
+        return 0;
+    }
+    auto offset = std::get_if<uint256_t>(&m.stack[0]);
+    auto buf = std::get_if<Buffer>(&m.stack[2]);
+    if (!buf || !offset) {
+        return 0;
+    }
+    if (*offset + inc > std::numeric_limits<uint64_t>::max()) {
```

Comment:
I think this check is a bit different than the one in Solidity. It doesn't check if offset + inc overflows a uint256, and it doesn't check m.stack[1]. If this check isn't identical to the checks done before deducting the variable gas in Solidity, it could cause a divergence.